### PR TITLE
aws-sdk

### PR DESCRIPTION
**Pull request by rclark**
Switches to aws-sdk for S3 requests and drops tilejson while trying to retain as much of the existing API as possible. There are a few things to notice:

The configuration object that you pass to the constructor must be one of 
- `{ data: { ... tilejson ...} }`
- `url.parse('s3://bucket/key-template')` (tiles only)
- `{ pathname: '... path to a tilejson file ...' }`

Some subtle adjustments to expected error cases that are part of the tests but may or may not be something that S3 will ever send (e.g. 503 with no body).

Drops the `._retry` function, opting instead for retries managed via aws-sdk. In order to consolidate this logic a bit, there's a function that defines a set of `ManagedError`s, and whether or not we want the aws-sdk to retry on those errors.

Drops the locking cache on credentials. aws-sdk is supposed to manage this internally.

Brings in a dependency on agentkeepalive, and uses its HTTPS client by default.

cc @willwhite @emilymcafee @yhahn

### CODE REVIEW

**Comment by rclark** [comment]
Also cc @ianshward -- we discussed error cases, timeouts and retries this morning, and I came out with this function that wraps any errors coming out of the aws-sdk: https://github.com/mapbox/tilelive-s3/blob/aws-sdk/lib/index.js#L79-L101

The goal here is to make it more transparent what the error cases are that we're handling / aware of and how we manage them.

File: lib/index.js [hunk 1]
```diff
@@ -28,39 +30,79 @@ function S3(uri, callback) {
     // If a uri of the form s3://[bucket]/[object] is passed,
     // generate implied HTTP resource and pass upstream for
     // retrieval to node-tilejson.
```
Comment (willwhite) [inline lib/index.js]: Is this comment/code still relevant now that tilejson is out of the picture?

Comment (rclark) [inline lib/index.js]: Yeah, good catch. Adjusted this, as it is still kind of confusing how we bounce from `s3://` to `https://` and back again

Comment (rclark) [inline lib/index.js]: Adjustment over here: https://github.com/mapbox/tilelive-s3/commit/67ad2cc9bc1fb7b7bb9b32d446596091e1803987

**Comment by rclark** [comment]
Closing in favor of #48

**Pull request closed without merging**